js/test-getting-data.js

```diff
@@ -0,0 +1,26 @@
+function testGettingData(data) {
+  if (!data) {
+    logError("unable to get data");
+    return false;
+  } else if (!data.currencyRates) {
+    logError("invalid data, unable to get data.currencyRates");
+    return false;
+  } else {
+    data.currencyRates.forEach((currencyObject, index) => {
+      if (
+        !currencyObject.base ||
+        !currencyObject.rates ||
+        !currencyObject.date ||
+        !currencyObject.timestamp
+      ) {
+        logError(`invalid data on index '${index}'`);
+        return false;

```

Comment:
For this test to work as you would expect, you need to use a `for` loop instead of `forEach`. The `return false` will only exit the `forEach` loop and go to the next operation, which is `return true`. So your `testGettingData` function will return `true` even though there was an error.